### PR TITLE
Format with black

### PR DESCRIPTION
**Pull request by ziw-liu**
Automatic formatting of `.py` documents with `black`.

Did not modify the auto-generated `./recOrder/plugin/qtdesigner/recOrder_ui.py`.

### CODE REVIEW

**Comment by ziw-liu** [comment]
'Initial formatting' is done manually here to break up the process into commits each modifying a single submodule for easier bug-fix traceback. This should not be necessary given the reliability of `black`, and is only performed out of an abundance of precaution. We should consider [enforcing formatting in our GItHub Actions workflow](https://black.readthedocs.io/en/stable/integrations/github_actions.html).

**Comment by ziw-liu** [comment]
This PR will start to address #176.

File: recOrder/plugin/widget/main_widget.py [hunk 1]
```diff
@@ -63,155 +79,225 @@ def __init__(self, napari_viewer: Viewer):
...
         self.ui.le_bg_path.editingFinished.connect(self.enter_acq_bg_path)
-        self.ui.qbutton_browse_bg_path.clicked[bool].connect(self.browse_acq_bg_path)
-        self.ui.qbutton_acq_birefringence.clicked[bool].connect(self.acq_birefringence)
+        self.ui.qbutton_browse_bg_path.clicked[bool].connect(
+            self.browse_acq_bg_path
+        )
+        self.ui.qbutton_acq_birefringence.clicked[bool].connect(
+            self.acq_birefringence
+        )
         self.ui.qbutton_acq_phase.clicked[bool].connect(self.acq_phase)
```
Comment (talonchandler) [inline recOrder/plugin/widget/main_widget.py:224]: Can we add line breaks to make these uniform?

File: recOrder/scripts/cli.py [hunk 2]
```diff
@@ -41,41 +43,106 @@ def help():
...
-@click.option('--overwrite', required=False, type=bool,
-              help='whether or not to overwrite any previous data under this name')
-def reconstruct(method, mode, data_dir, save_dir, name, config, overwrite=False):
+@click.help_option("-h", "--help")
+@click.option(
+    "--method",
+    required=False,
+    type=str,
+    help="mode of reconstruction: \
+                                                      QLIPP, IPS, UPTI",
```
Comment (talonchandler) [inline recOrder/scripts/cli.py:54]: ?

File: recOrder/scripts/cli.py [hunk 3]
```diff
@@ -41,41 +43,106 @@ def help():
...
+)
+@click.option(
+    "--mode",
+    required=False,
+    type=str,
+    help="mode of reconstruction: \
+                                                      2D, 3D, Stokes",
```
Comment (talonchandler) [inline recOrder/scripts/cli.py:61]: ?

**Comment by talonchandler** [comment]
Roger...I just took a quick flip through and highlighted some of the points that are really bizarre. 

Sorry I let this slip. Can you merge main into this branch, reformat, then I'll test Monday morning?

**Comment by talonchandler** [comment]
Fixed two typos I spotted in my sift through the changes.

File: recOrder/calib/Calibration.py [hunk 4]
```diff
@@ -730,50 +874,83 @@ def display_and_check_ROI(self, rect):
...
-            inst_mat = np.array([[1, 0, 0, -1],
-                                 [1, np.sin(2 * np.pi * chi), 0, -np.cos(2 * np.pi * chi)],
-                                 [1, -0.5 * np.sin(2 * np.pi * chi),
-                                  np.sqrt(3) * np.cos(np.pi * chi) * np.sin(np.pi * chi), -np.cos(2 * np.pi * chi)],
-                                 [1, -0.5 * np.sin(2 * np.pi * chi), -np.sqrt(3) / 2 * np.sin(2 * np.pi * chi),
-                                  -np.cos(2 * np.pi * chi)]])
+            inst_mat = np.array(
+                [
+                    [1, 0, 0, -1],
+                    [1, np.sin(2 * np.pi * chi), 0, -np.cos(2 * np.pi * chi)],
```
[the remark this answers is not in the document]
Comment (ziw-liu) [reply recOrder/calib/Calibration.py:924]: Sure!

File: recOrder/calib/Calibration.py [hunk 5]
```diff
@@ -1050,7 +1267,9 @@ def schnoor_fit_inv(retardance, a, b1, b2, c, d, e, wavelength):
 
         """
 
-        voltage = c * (((b1 + b2 / wavelength ** 2) / (retardance - a)) ** (1 / e) - 1) ** (1 / d)
+        voltage = c * (
+            ((b1 + b2 / wavelength**2) / (retardance - a)) ** (1 / e) - 1
+        ) ** (1 / d)
```
[the remark this answers is not in the document]
Comment (ziw-liu) [reply recOrder/calib/Calibration.py:1272]: Unfortunately not when you have `black` automatically handle it. Feel free to push your prefered edits though.

File: recOrder/compute/qlipp_compute.py [hunk 6]
```diff
@@ -74,76 +89,100 @@ def initialize_reconstructor(pipeline, image_dim=None, wavelength_nm=None, swing
...
-                             [1, np.sin(2 * np.pi * swing), 0, -np.cos(2 * np.pi * swing)],
-                             [1, -0.5 * np.sin(2 * np.pi * swing), np.sqrt(3) * np.cos(np.pi * swing) * np.sin(np.pi * swing),
-                              -np.cos(2 * np.pi * swing)],
-                             [1, -0.5 * np.sin(2 * np.pi * swing), -np.sqrt(3) / 2 * np.sin(2 * np.pi * swing),
-                              -np.cos(2 * np.pi * swing)]])
+    if calibration_scheme == "4-State":
+        inst_mat = np.array(
+            [
+                [1, 0, 0, -1],
+                [1, np.sin(2 * np.pi * swing), 0, -np.cos(2 * np.pi * swing)],
```
[the remark this answers is not in the document]
Comment (ziw-liu) [reply recOrder/compute/qlipp_compute.py:163]: Yes!

**Comment by ziw-liu** [comment]
This PR is getting increasingly cluttered with conflicts now. I'm tempted to close it and open a new one following [instructions here](https://black.readthedocs.io/en/stable/guides/introducing_black_to_your_project.html#avoiding-ruining-git-blame). Any suggestions @talonchandler?

**Comment by talonchandler** [comment]
I'm very in favor of preserving git-blame info if possible (I use this info for context quite often). 

Let's do this on Wednesday...I just sent you an invite.

**Comment by ziw-liu** [comment]
[GitHub's doc on formatter & `git blame`](https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view)

**Comment by talonchandler** [comment]
@ziw-liu and I are closing together. We will revisit black formatting after release.

**Pull request closed without merging**